crates/scheduler/src/call.rs

```diff
@@ -52,7 +54,7 @@ macro_rules! or_false {
 macro_rules! trap_use {
     ($($x:ident),* $(,)?) => {{
         $( let _ = $x; )*
-        Err(crate::Trap)?
+        Err(crate::Failure::Trap(crate::Trap))?
```

Comment:
I'm surprised this is needed. The conversion should be automatic and using `Err(crate::Trap)?` is more generic.